src/subnet_rental_canister/src/canister.rs

```diff
@@ -684,15 +685,18 @@ pub async fn refund() -> Result<u64, String> {
 /// Estimates how many cycles and days a given ICP amount would provide for a subnet rental.
 /// Uses the (potentially cached) exchange rate from the previous midnight to calculate the conversion.
 #[update]
-pub async fn subnet_topup_estimate(subnet_id: Principal, icp: Tokens) -> Result<TopupData, String> {
+pub async fn subnet_topup_estimate(
+    subnet_id: Principal,
+    icp: Tokens,
+) -> Result<TopUpSummary, String> {
     let Some(rental_agreement) = get_rental_agreement(&subnet_id) else {
         return Err("Rental agreement not found".to_string());
     };
 
     let now_secs = ic_cdk::api::time() / BILLION;
     let prev_midnight = round_to_previous_midnight(now_secs);
 
-    let Ok((scaled_exchange_rate_xdr_per_icp, _decimals)) =
+    let Ok((scaled_exchange_rate_xdr_per_icp, _decimals)) = // decimals is always 9 for XDR
```

Comment:
> decimals is always 9 for XDR

Is this really a GUARANTEED? Is this maybe just an incidental property of the current implementation, but it could be changed later? It seems safer (yet not much more difficult) to assume the latter.